content/posts/numpy/numpy2/index.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "18 years in the Making: NumPy 2.0 Released"
+date: 2024-06-16
+draft: false
+description: "
+18 years in the Making: NumPy 2.0 Released
+"
+tags: ["News", "numpy"]
+displayInList: true
+author: ["NumPy Developers"]
+---
+
+After 18 years since the release of NumPy 1.0, we are thrilled to announce the
+launch of NumPy 2.0! This major release marks a significant milestone in the
+evolution of NumPy, bringing a wealth of enhancements and improvements to users
+and unlocking future developments.
+
+NumPy has improved and evolved over the past 18 years and many old releases had
+significant performance, usability, and consistency improvements.
+Our approach for a long time has been to make incremental changes while
+carefully managing backward compatibility. This approach finally had to come to
+an end, for this one-off major release, with the significant improvements to
+NumPy's type system. The type system is fundamental to NumPy, and major
+behavioral changes could not be made incrementally, since mixing two different
+type systems would be a recipe for disaster.
+
+The journey to an actual 2.0 release has been long, and it was difficult to
+build the necessary momentum. In part this may be because for a time the idea
+of doing a NumPy 2.0 release among the NumPy developers required a revolutionary change with
+rewrites of significant key pieces in the code base. Many of these rewrites and
+changes happened over the years, but because of backward compatibility concerns
+they remained largely invisible to the users. NumPy 2.0 is, in part, the
+culmination of these improvements, allowing us to discard some of the legacy
+behaviors or details of the ABI that prevent future improvements.
+
+We started concrete plans for the 2.0 release more than a year ago, at a four hour
+long [public planning meeting](https://github.com/numpy/archive/tree/main/2.0_developer_meeting)
+in April 2023. Many of the key changes were proposed and discussed. The key goals
```

Comment:
It would be worth mentioning that the NumPy 2.0 feature planning started in 2019 already, with a significant part of the NumPy grant at BIDS being targeted at doing the footwork for the dtype refactoring and other related API cleanups. I think the above is correct in that the "NumPy 2.0" notion was only floated in 2023, but as described in the paragraphs above it is the culmination of a lot of other pieces of work.

Reply:
I'd prefer to not call out one grant or group in particular here. Some ideas went back much further (e.g., C API cleanups), some others came in only last year (e.g., Python API cleanup), some started in another place (e.g., array API standard started in 2020). NASA, CZI EOSS (2x) were also major grants, as were direct support from Quansight and NVIDIA.

Reply:
Re timelines: we could say something like "Some of the key features of this release and major changes to core NumPy machinery have been in the works for much longer, since 2019 at least". Would that work?

Reply:
Agreed that we don't need to call out specific grants (or, if we do, that we need to call out all). We just need the timeline to align with the work that happened.

Reply:
Or just "have been in the works since 2019 or even earlier."

Reply:
Calling out the grant could just be calling out the "first" grant, also.  No strong opinion, for me the April meeting isn't really the starting gun either, just the "formalizing" step.  (Not sure when a rough consenus slowly formed on it, I expect it was in the second half of 2022, really)

Reply:
I pushed a sentence. Added it as the first sentence in this paragraph - better than in the middle, because that breaks the flow of the story in this paragraph. Does that look good?